tests/unit/test_template.py

```diff
@@ -0,0 +1,105 @@
+        """
+        input = '2014-05-01T02:26:28Z'
+        output = format_date(input)
+        assert len(output) == 10
+        assert 'T' not in output
+        assert output == '2014-05-01'
+
+    def test_get_domain(self):
+        """Test domain name extraction from title.
+
+        (This really means give us the first pile of strings before
+        the first whitespace).
+        """
+        title = 'staging.webcompat.com - site is not usable'
+        assert get_domain(title) == 'staging.webcompat.com'
+        title = '😎 😓'
+        assert get_domain(title) == '😎'
+
+    def test_get_description(self):
+        """Test that we can pull out the first part of the description."""
+        body_html_frag = """
+<p><strong>Problem type</strong>: Site is not usable<br>
+<strong>Description</strong>: Missing items<br>
+<strong>Steps to Reproduce</strong>:<br>
+"""
+        body_html_frag2 = '😎'
+        self.assertEqual(get_description(body_html_frag), 'Missing items')
+        self.assertEqual(get_description(body_html_frag2), None)
+
+    def test_format_title(self):
+        """Test putting issue titles together."""
+        issue_data = {}
+        issue_data['title'] = 'staging.webcompat.com - desc from title'
+        issue_data['body_html'] = """
+<p><strong>Problem type</strong>: Site is not usable<br>
+<strong>Description</strong>: Desc from Description<br>
+<strong>Steps to Reproduce</strong>:<br>
+"""
+        formatted_title = format_title(issue_data)
+        self.assertEqual(formatted_title,
+                         'staging.webcompat.com - Desc from Description')
+        issue_data['body_html'] = 'fix my bug'
+        formatted_title = format_title(issue_data)
+        self.assertEqual(formatted_title,
+                         'staging.webcompat.com - desc from title')
+
+    def test_format_milestone_title(self):
+        """Test that we return a properly formatted milestone title."""
+        issue_data = {}
+        issue_data['milestone'] = {}
+        issue_data['milestone']['title'] = ''
+        issue_data['state'] = 'open'
+        self.assertEqual(format_milestone_title(issue_data),
+                         'Missing Milestone!')
+        issue_data['milestone']['title'] = 'sitewait'
+        self.assertEqual(format_milestone_title(issue_data),
+                         'Site Contacted')
+        issue_data['state'] = 'closed'
+        self.assertEqual(format_milestone_title(issue_data),
+                         'Closed: Site Contacted')
+
+    def test_format_milestone_class(self):
+        """Test that we return the correct class string."""
+        issue_data = {}
+        issue_data['milestone'] = {}
+        issue_data['milestone']['title'] = 'needsdiagnosis'
+        issue_data['state'] = 'open'
+        self.assertEqual(format_milestone_class(issue_data), 'needsdiagnosis')
+        issue_data['state'] = 'closed'
+        self.assertEqual(format_milestone_class(issue_data), 'closed')
```

Comment:
same logic than previous comment 

```suggestion
        # open state
        issue_data = {'milestone': {'title': 'needsdiagnosis'}, 'state': 'open'}
        assert format_milestone_class(issue_data) == 'needsdiagnosis'
        # closed state
        issue_data = {'milestone': {'title': 'needsdiagnosis'}, 'state': 'closed'}
        assert format_milestone_class(issue_data) == 'closed'
```